server/service/orbit.go

```diff
@@ -1065,36 +1066,34 @@ func (svc *Service) EscrowLUKSData(ctx context.Context, passphrase string, slotK
 	if clientError != "" {
 		return svc.ds.ReportEscrowError(ctx, host.ID, clientError)
 	}
-	encryptedPassphrase, encryptedSlotKey, err := svc.validateAndEncrypt(ctx, passphrase, slotKey)
+
+	encryptedPassphrase, encryptedSalt, validatedKeySlot, err := svc.validateAndEncrypt(ctx, passphrase, salt, keySlot)
 	if err != nil {
 		_ = svc.ds.ReportEscrowError(ctx, host.ID, err.Error())
 		return err
 	}
 
-	return svc.ds.SaveLUKSData(ctx, host.ID, encryptedPassphrase, encryptedSlotKey)
+	return svc.ds.SaveLUKSData(ctx, host.ID, encryptedPassphrase, encryptedSalt, validatedKeySlot)
 }
 
-func (svc *Service) validateAndEncrypt(ctx context.Context, passphrase string, slotKey string) (string, string, error) {
-	if passphrase == "" {
-		return "", "", badRequest("Blank passphrase provided")
+func (svc *Service) validateAndEncrypt(ctx context.Context, passphrase string, salt string, keySlot *uint) (string, string, uint, error) {
```

Comment:
Nit, when having many same-typed return types, it's a good practice to use named returns:
`(string, string, uint, error)` -> `(encryptedPassphrase string, encryptedSalt string, keySlot uint, err error)`

Reply:
Thanks for the call-out. Will fix.